### ISSUE TITLE
After Updating into VS Code 1.86, facing Error - "Resolver error: Error: The VS Code Server failed to start"

### ISSUE THREAD

**Issue by anujkrsharmax86** [microsoft/vscode-remote-release#9524]
We have written the needed data into your clipboard because it was too large to send. Please paste.
Type: <b>Bug</b>

When i connect with remote host using Remote-SSH extension. Facing below issue which is appearing in output log - 

[13:08:26.676] Log Level: 2
[13:08:26.682] VS Code version: 1.86.0
[13:08:26.682] Remote-SSH version: remote-ssh@0.109.2024020215
[13:08:26.682] darwin arm64
[13:08:26.683] SSH Resolver called for "ssh-remote+<remote-host-name>", attempt 1
[13:08:26.683] "remote.SSH.useLocalServer": true
[13:08:26.683] "remote.SSH.useExecServer": false
[13:08:26.683] "remote.SSH.path": undefined
[13:08:26.683] "remote.SSH.configFile": undefined
[13:08:26.683] "remote.SSH.useFlock": true
[13:08:26.683] "remote.SSH.lockfilesInTmp": false
[13:08:26.684] "remote.SSH.localServerDownload": auto
[13:08:26.684] "remote.SSH.remoteServerListenOnSocket": false
[13:08:26.684] "remote.SSH.showLoginTerminal": false
[13:08:26.684] "remote.SSH.defaultExtensions": []
[13:08:26.684] "remote.SSH.loglevel": 2
[13:08:26.684] "remote.SSH.enableDynamicForwarding": true
[13:08:26.684] "remote.SSH.enableRemoteCommand": false
[13:08:26.684] "remote.SSH.serverPickPortsFromRange": {}
[13:08:26.684] "remote.SSH.serverInstallPath": {}
[13:08:26.687] SSH Resolver called for host: <remote-host-name>
[13:08:26.687] Setting up SSH remote "<remote-host-name>"
[13:08:26.689] Acquiring local install lock: /var/folders/r6/p1w4rs854xz6_bfn29sssbl40000gn/T/vscode-remote-ssh-fafc4cdb-install.lock
[13:08:26.689] Looking for existing server data file at /Users/<user>/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-fafc4cdb-05047486b6df5eb8d44b2ecd70ea3bdf775fd937-0.109.2024020215-tr/data.json
[13:08:26.689] Using commit id "05047486b6df5eb8d44b2ecd70ea3bdf775fd937" and quality "stable" for server
[13:08:26.691] Install and start server if needed
[13:08:26.693] PATH: /Users/<user>/.rd/bin:/usr/local/bin:/Users/<user>/codecommit/boss/BOSS_SETUP/tools/javacli/bin:/opt/homebrew/bin:/opt/homebrew/Cellar/maven/3.8.6/libexec/bin:/Library/Java/JavaVirtualMachines/jdk-17.0.4.1.jdk/Contents/Home/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/usr/local/share/dotnet
[13:08:26.693] Checking ssh with "ssh -V"
[13:08:26.703] > OpenSSH_9.4p1, LibreSSL 3.3.6

[13:08:26.705] askpass server listening on /var/folders/r6/p1w4rs854xz6_bfn29sssbl40000gn/T/vscode-ssh-askpass-c24d4bf2f823e05c6e63a181e98afda1eddebcb1.sock
[13:08:26.705] Spawning local server with {"serverId":1,"ipcHandlePath":"/var/folders/r6/p1w4rs854xz6_bfn29sssbl40000gn/T/vscode-ssh-askpass-5f36ab4d526770ddefd6977454290b55a6f088a1.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","52780","-o","ConnectTimeout=15","<remote-host-name>"],"serverDataFolderName":".vscode-server","dataFilePath":"/Users/<user>/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-fafc4cdb-05047486b6df5eb8d44b2ecd70ea3bdf775fd937-0.109.2024020215-tr/data.json"}
[13:08:26.705] Local server env: {"SSH_AUTH_SOCK":"/private/tmp/com.apple.launchd.il6TjqO4AK/Listeners","SHELL":"/bin/bash","DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin)","VSCODE_SSH_ASKPASS_EXTRA_ARGS":"","VSCODE_SSH_ASKPASS_MAIN":"/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/r6/p1w4rs854xz6_bfn29sssbl40000gn/T/vscode-ssh-askpass-c24d4bf2f823e05c6e63a181e98afda1eddebcb1.sock"}
[13:08:26.707] Spawned 31378
[13:08:26.821] > local-server-1> Running ssh connection command: ssh -v -T -D 52780 -o ConnectTimeout=15 <remote-host-name>
[13:08:26.823] > local-server-1> Spawned ssh, pid=31387
[13:08:26.872] stderr> OpenSSH_9.4p1, LibreSSL 3.3.6
[13:08:27.411] stderr> debug1: Server host key: ssh-ed25519 SHA256:TPVXC7s0chwj26mi01jk9/N/ITKkIrf2l/snhxFnZTE
[13:08:27.912] stderr> Authenticated to <remote-host-name> ([100.69.211.107]:22) using "publickey".
[13:08:28.201] > ready: cbeb43f9e74a
[13:08:28.270] > Linux 5.4.17-2136.324.5.3.el7uek.x86_64 #2 SMP Tue Oct 10 12:44:19 PDT 2023
[13:08:28.270] Platform: linux
[13:08:28.342] > /bin/ksh
[13:08:28.342] Shell: ksh
[13:08:28.342] Creating bash subshell inside "ksh"
[13:08:28.410] > bash version:  4.2.46(2)-release
[13:08:28.463] > bash version:  4.2.46(2)-release
[13:08:28.564] > cbeb43f9e74a: running
[13:08:28.601] > Missing GLIBCXX >= 3.4.25!
> Found versions 3.4.1
> 3.4.2
> 3.4.3
> 3.4.4
> 3.4.5
> 3.4.6
> 3.4.7
> 3.4.8
> 3.4.9
> 3.4.10
> 3.4.11
> 3.4.12
> 3.4.13
> 3.4.14
> 3.4.15
> 3.4.16
> 3.4.17
> 3.4.18
> 3.4.19
[13:08:28.620] > Missing GLIBC >= 2.28!
> Found version ldd (GNU libc) 2.17
> cbeb43f9e74a$$1$$
> Acquiring lock on /home/<user>/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/vscode-remote-lock.<user>.05047486b6df5eb8d44b2ecd70ea3bdf775fd937
> Found existing installation at /home/<user>/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937...
> Checking /home/<user>/.vscode-server/.05047486b6df5eb8d44b2ecd70ea3bdf775fd937.log and /home/<user>/.vscode-server/.05047486b6df5eb8d44b2ecd70ea3bdf775fd937.pid for a running server
> Looking for server with pid: 6822
[13:08:28.659] > Starting VS Code Server... "/home/<user>/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/bin/code-server" --start-server --host=127.0.0.1 --accept-server-license-terms --enable-remote-auto-shutdown --port=0 --telemetry-level all   &> "/home/<user>/.vscode-server/.05047486b6df5eb8d44b2ecd70ea3bdf775fd937.log" < /dev/null
> The VS Code Server log file is /home/<user>/.vscode-server/.05047486b6df5eb8d44b2ecd70ea3bdf775fd937.log
> printenv:
[13:08:28.659] >     XDG_SESSION_ID=48913
>     HOSTNAME=mumbai237673
>     GUESTFISH_INIT=\e[1;34m
>     SHELL=/bin/ksh
>     HISTSIZE=1000
>     SSH_CLIENT=10.213.204.101 52782 22
>     QTDIR=/usr/lib64/qt-3.3
>     QTINC=/usr/lib64/qt-3.3/include
>     QT_GRAPHICSSYSTEM_CHECKED=1
>     _AST_FEATURES=UNIVERSE - ucb
>     USER=<user>
>     ADE_AUTO_MAP=disabled
>     ADE_DEFAULT_VIEW_STORAGE_LOC=/scratch/<user>/view_storage
>     ADE_CHROOT_SHELL=/usr/local/packages/utl_main_linux/LATEST/utl/chroot_env.pl
>     VSCODE_AGENT_FOLDER=/home/<user>/.vscode-server
>     A__z="*SHLVL
>     GUESTFISH_PS1=\[\e[1;32m\]><fs>\[\e[0;31m\] 
>     PATH=/usr/dev_infra/platform/bin:/usr/dev_infra/generic/bin:/usr/lib64/qt-3.3/bin:/bin:/usr/bin:/usr/local/bin:/usr/X11R6/bin:/usr/local/ade/bin
>     MAIL=/var/spool/mail/<user>
>     ADE_DEFAULT_TWORK_STORAGE_LOC=/scratch/<user>/view_storage
>     ADE_CONFIG_DIR=/ade_autofs/ade_infra/SEICONFIG_MAIN_PLATFORMS.rdd/LATEST/seiconfig/ade
>     PWD=/home/<user>
>     INPUTRC=/etc/inputrc
>     LANG=en_US.UTF-8
>     MODULEPATH=/usr/share/Modules/modulefiles:/etc/modulefiles
>     TZ=Asia/Calcutta
>     LOADEDMODULES=
>     KDEDIRS=/usr
>     GUESTFISH_OUTPUT=\e[0m
>     SSH_ASKPASS=/usr/libexec/openssh/gnome-ssh-askpass
>     SHLVL=2
>     HOME=/home/<user>
>     LOGNAME=<user>
>     QTLIB=/usr/lib64/qt-3.3/lib
>     CVS_RSH=ssh
>     XDG_DATA_DIRS=/home/<user>/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share
>     SSH_CONNECTION=10.213.204.101 52782 100.69.211.107 22
>     MODULESHOME=/usr/share/Modules
>     LESSOPEN=||/usr/bin/lesspipe.sh %s
>     GCONF_LOCAL_LOCKS=1
>     XDG_RUNTIME_DIR=/run/user/747178
>     QT_PLUGIN_PATH=/usr/lib64/kde4/plugins:/usr/lib/kde4/plugins
>     GUESTFISH_RESTORE=\e[0m
>     _=/bin/printenv
> Removing old logfile at /home/<user>/.vscode-server/.05047486b6df5eb8d44b2ecd70ea3bdf775fd937.log
[13:08:28.681] > Spawned remote server: 10385
> Waiting for server log...
[13:08:28.691] > Waiting for server log...
[13:08:28.729] > Waiting for server log...
[13:08:28.801] > Waiting for server log...
[13:08:28.844] > Waiting for server log...
[13:08:28.884] > Waiting for server log...

...         ...        ...

[13:08:42.814] > Waiting for server log...
[13:08:42.854] > Waiting for server log...
[13:08:42.889] > Waiting for server log...
[13:08:42.927] > Waiting for server log...
[13:08:42.963] > Waiting for server log...
[13:08:43.021] >  
> *
> * Visual Studio Code Server
> *
> * By using the software, you agree to
> * the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-license) and
> * the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacystatement).
> *
>  
[13:08:43.022] > Server did not start successfully. Full server log at /home/<user>/.vscode-server/.05047486b6df5eb8d44b2ecd70ea3bdf775fd937.log >>>
> Warning: Missing GLIBCXX >= 3.4.25! from /usr/lib64/libstdc++.so.6.0.19
> Warning: Missing GLIBC >= 2.28! from /usr/lib64/libc-2.17.so
> Error: Missing required dependencies. Please refer to our FAQ https://aka.ms/vscode-remote/faq/old-linux for additional information.
> <<< End of server log
> cbeb43f9e74a: start
> exitCode==32==
> listeningOn====
> osReleaseId==ol==
> arch==x86_64==
> vscodeArch==x64==
> bitness==64==
> tmpDir==/run/user/747178==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime==14359==
> cbeb43f9e74a: end
[13:08:43.023] Received install output: 
exitCode==32==
listeningOn====
osReleaseId==ol==
arch==x86_64==
vscodeArch==x64==
bitness==64==
tmpDir==/run/user/747178==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
extInstallTime====
serverStartTime==14359==

[13:08:43.025] Terminating local server
[13:08:43.027] Resolver error: Error: The VS Code Server failed to start
	at g.ServerInstallError (/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:460179)
	at d (/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:453522)
	at t.handleInstallOutput (/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:455792)
	at e (/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:518696)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
	at async /Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:540605
	at async t.withShowDetailsEvent (/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:544264)
	at async /Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:515401
	at async T (/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:513452)
	at async t.resolveWithLocalServer (/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:514941)
	at async k (/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:537822)
	at async t.resolve (/Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:541635)
	at async /Users/<user>/.vscode/extensions/ms-vscode-remote.remote-ssh-0.109.2024020215/out/extension.js:2:807803
[13:08:43.030] ------

[13:08:43.033] Local server exit: 15

Extension version: 0.109.2024020215
VS Code version: Code 1.86.0 (05047486b6df5eb8d44b2ecd70ea3bdf775fd937, 2024-01-31T10:29:11.933Z)
OS version: Darwin arm64 23.3.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|5, 5, 5|
|Memory (System)|32.00GB (0.15GB free)|
|Process Argv|--crash-reporter-id 1295b0e0-0d64-4e32-97e3-f362a984ed40|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30951796
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementsc:30887149
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
dsvsc019bcf:30953938
3ef8e399:30949928
e3gdj431:30958358

```

</details>

**Comment by BesjanTomja** [comment]
same

**Comment by idchlife** [comment]
New rules, new versions required on the server:

https://code.visualstudio.com/docs/remote/linux#_remote-host-container-wsl-linux-prerequisites

Either downgrade vscode + remote ssh extension or upgrade your server.

For me neither was an option so I just switched to SSH FS extension instead of this one.

**Comment by connor4312** [comment]
This is fixed in VS Code 1.86.1.

**Comment by anujkrsharmax86** [comment]
@connor4312 thanks. Just now tested with 1.86.1, it is working. 

Bug fix list for 1.86.1 also contains this - 
https://github.com/microsoft/vscode/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22December+%2F+January+2024+Recovery+1%22
